### ISSUE TITLE
Any recommendation for supported camera product?

### ISSUE THREAD

**Issue by scaret** [agsh/onvif#7]
Thanks for this library!

Recently I'm playing with camera. My foscam supports part of ONVIF but pretty incomplete. Do you have some recommendation for cameras that fully supports the protocol?

**Comment by agsh** [comment]
Hi!
I've never used foscam before. So what does incomplete ONVIF support means? Maybe it's just buggy server realization of the protocol? Or maybe troubles in client (this library, for example, doesn't complete all ONVIF specification, just a few sections: device, discovery, ptz, media, events).
Did you try ONVIF Device Manager(http://sourceforge.net/projects/onvifdm/) with foscam?

**Comment by scaret** [comment]
Thank you for your reply.

Incomplete ONVIF support means that my camera will have no response to some of the methods of node-ONVIF. And some of the functionality just give the wrong response.

For example, `absoluteMove` works fine, so I assume that foscam hardware have the ability to get/set the current camera direction. But when it come to the function `getStatus`, the position field will always return zero.
```
 {
        "position": {
            "x": 0,
            "y": 0,
            "zoom": 0
        },
        "moveStatus": {
            "panTilt": "IDLE",
            "zoom": "IDLE"
        },
        "utcTime": 1431933429
    }
     {
        "position": {
            "x": 0,
            "y": 0,
            "zoom": 0
        },
        "moveStatus": {
            "panTilt": "MOVING",
            "zoom": "IDLE"
        },
        "utcTime": 1431933538
}
```

**Comment by agsh** [comment]
Oh, now I understand.
Sorry but I can't help you to choose proper ONVIF device cause I get only one camera by myself. Here it is: http://www.360visiontechnology.com/Products/Product-Page/Predator---IR-Thermal. It starts support ONVIF protocol recently. And still hasn't got events support. So a lot of work on ONVIF implementation in javascript I can do only with user's feedback.
By this experience I can say that problems may be not in foscam but in this library. So if something works in ONVIF device manager, but doesn't work here, you know what to do - write issues :)
To find out what happens you can use third argument of every callback function which represents raw SOAP response from the device.
Also try to run getCapabilities method to find out what things camera can support. I've tried to describe it in [documentation](http://agsh.github.io/onvif/Cam.html#~Capabilities) which is a copy of wsdl ONVIF [scheme](http://www.onvif.org/onvif/ver10/device/wsdl/devicemgmt.wsdl#op.GetCapabilities) with some practical additions.
And for rtsp streaming I've wrote a little wrapper for ffmpeg, maybe it will be useful: https://github.com/agsh/rtsp-ffmpeg

**Comment by scaret** [comment]
Thank you. 

I have looked into the SOAP message, and for the foscam ONVIF support, most problem is still on its implementation. For example, it claims that it supports the position information in capability list, but always returns 0. Another example is for the `getSystemDateAndTime`, its timestamp from SOAP message is a fixed number in 1980s, which is obviously foscam's problem.

That's fine for the device recommendation.

Noted the rstp-ffmpeg lib with thanks. Actually I have played with it before, but didn't realize it's author and ONVIF's is the same person. Good job! 

I have similar need for rstp-ffmpeg, but I won't use motion jpeg as the output. The source code is really helpful for ffmpeg usage, so thanks agan.
